devicetypes/fibargroup/fibaro-double-switch-2-zw5.src/fibaro-double-switch-2-zw5.groovy

```diff
@@ -0,0 +1,667 @@
+					type: "paragraph",
+					element: "paragraph"
+			)
+
+			input (
+					name: it.key,
+					title: null,
+					description: "Default: $it.def" ,
+					type: it.type,
+					options: it.options,
+					range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+					defaultValue: it.def,
+					required: false
+			)
+		}
+
+		input ( name: "logging", title: "Logging", type: "boolean", required: false )
+	}
+}
+
+
+def on(){
+	def cmds = []
+	cmds << [zwave.basicV1.basicSet(value: 255), 1]
+	cmds << [zwave.switchBinaryV1.switchBinaryGet(),1]
+	encapSequence(cmds, 5000)
+}
+
+def off(){
+	def cmds = []
+	cmds << [zwave.basicV1.basicSet(value: 0), 1]
+	cmds << [zwave.switchBinaryV1.switchBinaryGet(),1]
+	encapSequence(cmds, 5000)
+}
+
+def childOn() {
+	sendHubCommand(response(encap(zwave.basicV1.basicSet(value: 255),2)))
+	sendHubCommand(response(encap(zwave.switchBinaryV1.switchBinaryGet(),2)))
+}
+
+def childOff() {
+	sendHubCommand(response(encap(zwave.basicV1.basicSet(value: 0),2)))
+	sendHubCommand(response(encap(zwave.switchBinaryV1.switchBinaryGet(),2)))
+}
+
+def reset() {
+	def cmds = []
+	cmds << [zwave.meterV3.meterReset(), 1]
+	cmds << [zwave.meterV3.meterGet(scale: 0), 1]
+	cmds << [zwave.meterV3.meterGet(scale: 2), 1]
+	encapSequence(cmds,1000)
+}
+
+def childReset() {
+	def cmds = []
+	cmds << response(encap(zwave.meterV3.meterReset(), 2))
+	cmds << response(encap(zwave.meterV3.meterGet(scale: 0), 2))
+	cmds << response(encap(zwave.meterV3.meterGet(scale: 2), 2))
+	sendHubCommand(cmds,1000)
+}
+
+def refresh() {
+	def cmds = []
+	cmds << [zwave.meterV3.meterGet(scale: 0), 1]
+	cmds << [zwave.meterV3.meterGet(scale: 2), 1]
+	cmds << [zwave.switchBinaryV1.switchBinaryGet(),1]
+	encapSequence(cmds,1000)
+}
+
+def childRefresh() {
+	def cmds = []
+	cmds << response(encap(zwave.meterV3.meterGet(scale: 0), 2))
+	cmds << response(encap(zwave.meterV3.meterGet(scale: 2), 2))
+	cmds << response(encap(zwave.switchBinaryV1.switchBinaryGet(), 2))
+	sendHubCommand(cmds,1000)
+}
+
+def setState(String key, Integer value) {
+	state."$key".value = value
+}
+
+Integer getState(String key) {
+	state."$key".value
+}
+
+def installed(){
+	sendEvent(name: "checkInterval", value: 1920, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	initialize()
+	response(refresh())
+}
+
+def ping() {
+	refresh()
+}
+
+//Configuration and synchronization
+def updated() {
+	if ( state.lastUpdated && (now() - state.lastUpdated) < 500 ) return
+	initialize()
+}
+
+def initialize() {
+	def cmds = []
+	logging("${device.displayName} - Executing initialize()","info")
+	if (!childDevices) {
+		createChildDevices()
+	}
+	if (device.currentValue("numberOfButtons") != 6) { sendEvent(name: "numberOfButtons", value: 6) }
+
+	cmds << zwave.multiChannelAssociationV2.multiChannelAssociationGet(groupingIdentifier: 1) //verify if group 1 association is correct  
+	cmds << [zwave.switchBinaryV1.switchBinaryGet(),1]

```

Comment:
This get is already getting called in refresh. I think it's redundant here.

Reply:
@greens  This was added because in OneApp we were not getting the correct initial state.

Reply:
What's the actual cause of that, though? It would be better to figure out the actual reason we're failing to get initial state and fix that than just polling twice.

Reply:
@greens cmds << [zwave.switchBinaryV1.switchBinaryGet(),1] has been removed from def initialize().

Reply:
Does it still get initial state in OneApp?

Reply:
Yes. After I made the change I made sure to test this in OneApp.